scripts/test-many-configs.sh

```diff
@@ -0,0 +1,419 @@
+                flag=1
+                return 1
+            else
+                let "x = $numFlakyTests - $expectedTests"
+                echo "There were $x more tests found than expected in ${testNum}. %%%%%"
+                flag=1
+                return 1
+           fi
+        fi
+    fi
+}
+
+
+
+function checkTimeout() {
+    testNum=$1
+    expectedTime=$2
+    if [[ ! -f ${testNum} ]]; then
+        return 0
+    fi
+
+    if grep "Using a timeout of ${expectedTime}" ${testNum}.log; then
+        echo "Success: ${testNum} used a timeout of ${expectedTime} as expected."
+        return 0
+    else
+        echo "Error: ${testNum} did not use a timeout. %%%%%"
+        flag=1
+        return 1
+    fi
+}
+
+
+
+
+while IFS="," read -r URL SHA MODULE testCount1 testCount2 testCount3 testCount4 testCount5 testCount6 testCount7 testCount8 testCount9; do
+
+
+    renamedRepo=${URL}"/"
+    readarray -d / -t starr <<< "${renamedRepo}"
+
+
+    #1. Clone the project.
+
+    if [[ ! -d ${starr[4]} ]]; then
+        git clone ${URL}.git ${starr[4]}
+    fi
+
+
+
+
+    #2. Navigate to proj and checkout SHA
+
+    cd ${starr[4]}
+    projectDirectory=$(pwd)
+    git checkout ${SHA}
+
+
+
+
+    #3. Modify pom file
+
+    git checkout -f .
+    bash ${scriptDir}/../pom-modify/modify-project.sh ${projectDirectory} 1.2.0-SNAPSHOT
+
+
+
+
+    #4 Maven install the proj
+
+    if [[ ${MODULE} != "" ]]; then
+        PL="-pl ${MODULE}"
+    else
+        PL=""
+    fi
+
+    if [[ $URL == "https://github.com/wildfly/wildfly" ]]; then
+        sed -i 's;<url>http://repository.jboss.org/nexus/content/groups/public/</url>;<url>https://repository.jboss.org/nexus/content/groups/public/</url>;' pom.xml
+    fi
+
+    mvn install -DskipTests ${PL} -am -B -q
+    if [[ $? != 0 ]]; then
+        echo "Project ${starr[4]} was not installed successfully.  %%%%%"
+        flag=1
+    else
+
+        #5. Run tests
+
+        echo "Start of logs for ${projectDirectory}"
+
+        { time -p timeout 1h mvn test -fn -B ${PL} |& tee -a ${projectDirectory}/${MODULE}/mvn-test.log ;} 2> ${projectDirectory}/${MODULE}/mvn-test-time.log
+        cp ${projectDirectory}/${MODULE}/mvn-test.log ${projectDirectory}/mvn-test.log
+        cp ${projectDirectory}/${MODULE}/mvn-test-time.log ${projectDirectory}/mvn-test-time.log
+
+
+        rounds="-Ddt.randomize.rounds="
+        ogOrderPass="-Ddt.detector.original_order.all_must_pass="
+        detType="-Ddetector.detector_type="
+        countFirstFail="-Ddt.detector.count.only.first.failure="
+        verRounds="-Ddt.verify.rounds="
+        time="-Ddetector.timeout="
+        semantics="-Ddt.detector.roundsemantics.total="
+        absPath="-Ddt.cache.absolute.path="
+
+
+
+        setOriginalOrder ${starr[4]} ${MODULE}
+        #Test 1: Simply test if the timeout configuration is working. Not testing accuracy here since it has proven to be nondeterministic
```

Comment:
Propagate the same options for running testrunner from the other script to relevant places in this script.